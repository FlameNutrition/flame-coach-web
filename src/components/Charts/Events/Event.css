.complex-swipeable-list-item {
    background-color: white;
    padding: 10px;
    flex: 1;
    min-width: 0;
    user-select: none;
    cursor: pointer;
}

.complex-swipeable-list-item-label {
    display: flex;
    align-items: center;
}

.complex-swipeable-list-item-name {
    font-weight: bold;
    font-family: "Roboto", "Helvetica", "Arial", sans-serif;
    font-size: 19px;
    line-height: 27px;
    color: #162473;
    margin-left: 4px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

.complex-swipeable-list-item-description {
    margin-left: 2px;
    margin-top: 4px;
    color: #000000;
    font-family: "Roboto", "Helvetica", "Arial", sans-serif;
    font-size: 14px;
    line-height: 17px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}
