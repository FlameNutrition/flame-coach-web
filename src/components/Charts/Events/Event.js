import React from 'react';
import './Event.css';
import PropTypes from 'prop-types';
import { Paper } from '@material-ui/core';
import moment from 'moment';

const Event = ({
  date, value, measureType
}) => {
  return (
    <Paper
      className="complex-swipeable-list-item"
      variant="outlined"
      square
    >
      <div className="complex-swipeable-list-item-label">
        <span className="complex-swipeable-list-item-name">
          {value}
          {' '}
          {measureType}
        </span>
      </div>

      <div className="complex-swipeable-list-item-description">
        {moment(date).format('ddd, MMMM Do YYYY')}
      </div>

    </Paper>
  );
};

Event.propTypes = {
  date: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired,
  measureType: PropTypes.string
};

Event.defaultProps = {
  measureType: 'kg'
};

export default Event;
