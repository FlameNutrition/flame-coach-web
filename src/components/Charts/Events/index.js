import {
  Card, CardContent, makeStyles, Paper, Typography
} from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';
import { SwipeableList, SwipeableListItem } from '@sandstreamdev/react-swipeable-list';
import Event from './Event';
import clsx from 'clsx';

const useStyles = makeStyles(() => ({
  root: {
  },
  eventsCardTitle: {
    textAlign: 'center',
    paddingBottom: '5px'
  },
  complexSwipeableListContainer: {
    width: '100%',
    height: '90%',
    overflowY: 'scroll'
  },
  complexSwipeableListItemContentLeft: {
    backgroundColor: 'red',
    color: 'white',
    flex: 1,
    height: '100%',
    display: 'flex',
    flexFlow: 'column',
    justifyContent: 'center'
  },
  complexSwipeableListItemContentLeftText: {
    paddingLeft: '10px'
  }
}));

const Events = ({
  dataEvents,
  onDeleteHandle,
  measureUnit,
  className
}) => {
  const classes = useStyles();

  const swipeableListItemContentLeft = (
    <Paper className={classes.complexSwipeableListItemContentLeft} variant="outlined" square>
      <Typography className={classes.complexSwipeableListItemContentLeftText} variant="body1">
        Delete
      </Typography>
    </Paper>
  );

  return (
    <Card className={clsx(classes.root)}>
      <CardContent className={className}>
        <Typography
          className={classes.eventsCardTitle}
          component="h2"
          variant="h5"
          gutterBottom
        >
          Events
        </Typography>

        <div className={classes.complexSwipeableListContainer}>
          <SwipeableList threshold={0.65}>
            {dataEvents.reverse().map((event) => {
              return (
                <SwipeableListItem
                  key={event.identifier}
                  swipeRight={{
                    content: swipeableListItemContentLeft,
                    action: () => {
                      onDeleteHandle(event);
                    }
                  }}
                >
                  <Event date={event.date} value={event.value} measureType={measureUnit} />
                </SwipeableListItem>
              );
            })}
          </SwipeableList>
        </div>
      </CardContent>
    </Card>
  );
};

Events.propTypes = {
  className: PropTypes.string,
  dataEvents: PropTypes.array.isRequired,
  onDeleteHandle: PropTypes.func.isRequired,
  measureUnit: PropTypes.string.isRequired
};

Events.defaultProps = {
};

export default Events;
