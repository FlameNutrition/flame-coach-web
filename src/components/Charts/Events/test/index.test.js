import React from 'react';
import Events from '../index';
import { render, screen } from '../../../../testing/test-utils';
import sinon from 'sinon';

describe('<Events />', () => {
  const deleteHandler = sinon.spy();

  it('create component Events', () => {
    const events = [
      { identifier: 1, date: '2021-04-01', value: 50.5 },
      { identifier: 2, date: '2021-05-01', value: 70.5 },
      { identifier: 3, date: '2021-06-01', value: 80.5 }
    ];

    const { container } = render(<Events
      onDeleteHandle={deleteHandler}
      dataEvents={events}
    />);

    expect(container).toMatchSnapshot();
  });

  it('check events display', () => {
    const events = [
      { identifier: 1, date: '2021-04-01', value: 50.5 },
      { identifier: 2, date: '2021-05-01', value: 70.5 },
      { identifier: 3, date: '2021-06-01', value: 80.5 }
    ];

    render(<Events
      onDeleteHandle={deleteHandler}
      dataEvents={events}
    />);

    expect(screen.queryByText('Events')).not.toBeNull();

    expect(screen.queryByText('Thu, April 1st 2021')).not.toBeNull();
    expect(screen.queryByText('50.5 kg')).not.toBeNull();
    expect(screen.queryByText('Sat, May 1st 2021')).not.toBeNull();
    expect(screen.queryByText('70.5 kg')).not.toBeNull();
    expect(screen.queryByText('Tue, June 1st 2021')).not.toBeNull();
    expect(screen.queryByText('80.5 kg')).not.toBeNull();

    expect(screen.queryAllByText('Delete')).toHaveLength(3);
  });
});
