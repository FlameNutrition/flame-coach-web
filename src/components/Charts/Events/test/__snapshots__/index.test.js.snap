// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Events /> create component Events 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <h2
        class="MuiTypography-root makeStyles-eventsCardTitle-2 MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Events
      </h2>
      <div
        class="makeStyles-complexSwipeableListContainer-3"
      >
        <div
          class="swipeable-list"
        >
          <div
            class="swipeable-list-item"
          >
            <div
              class="swipeable-list-item__content-right"
              data-testid="swipe-right-content"
            >
              <div
                class="MuiPaper-root makeStyles-complexSwipeableListItemContentLeft-4 MuiPaper-outlined"
              >
                <p
                  class="MuiTypography-root makeStyles-complexSwipeableListItemContentLeftText-5 MuiTypography-body1"
                >
                  Delete
                </p>
              </div>
            </div>
            <div
              class="swipeable-list-item__content"
              data-testid="content"
            >
              <div
                class="MuiPaper-root complex-swipeable-list-item MuiPaper-outlined"
              >
                <div
                  class="complex-swipeable-list-item-label"
                >
                  <span
                    class="complex-swipeable-list-item-name"
                  >
                    80.5
                     
                    kg
                  </span>
                </div>
                <div
                  class="complex-swipeable-list-item-description"
                >
                  Tue, June 1st 2021
                </div>
              </div>
            </div>
          </div>
          <div
            class="swipeable-list-item"
          >
            <div
              class="swipeable-list-item__content-right"
              data-testid="swipe-right-content"
            >
              <div
                class="MuiPaper-root makeStyles-complexSwipeableListItemContentLeft-4 MuiPaper-outlined"
              >
                <p
                  class="MuiTypography-root makeStyles-complexSwipeableListItemContentLeftText-5 MuiTypography-body1"
                >
                  Delete
                </p>
              </div>
            </div>
            <div
              class="swipeable-list-item__content"
              data-testid="content"
            >
              <div
                class="MuiPaper-root complex-swipeable-list-item MuiPaper-outlined"
              >
                <div
                  class="complex-swipeable-list-item-label"
                >
                  <span
                    class="complex-swipeable-list-item-name"
                  >
                    70.5
                     
                    kg
                  </span>
                </div>
                <div
                  class="complex-swipeable-list-item-description"
                >
                  Sat, May 1st 2021
                </div>
              </div>
            </div>
          </div>
          <div
            class="swipeable-list-item"
          >
            <div
              class="swipeable-list-item__content-right"
              data-testid="swipe-right-content"
            >
              <div
                class="MuiPaper-root makeStyles-complexSwipeableListItemContentLeft-4 MuiPaper-outlined"
              >
                <p
                  class="MuiTypography-root makeStyles-complexSwipeableListItemContentLeftText-5 MuiTypography-body1"
                >
                  Delete
                </p>
              </div>
            </div>
            <div
              class="swipeable-list-item__content"
              data-testid="content"
            >
              <div
                class="MuiPaper-root complex-swipeable-list-item MuiPaper-outlined"
              >
                <div
                  class="complex-swipeable-list-item-label"
                >
                  <span
                    class="complex-swipeable-list-item-name"
                  >
                    50.5
                     
                    kg
                  </span>
                </div>
                <div
                  class="complex-swipeable-list-item-description"
                >
                  Thu, April 1st 2021
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
